tests/unit_tests/instancetype_test.go

```diff
@@ -0,0 +1,178 @@
+package tests
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+)
+
+const (
+	expectedVersion          = "instancetype.kubevirt.io/v1beta1"
+	expectedVendor           = "kubevirt.io"
+	instanceTypeErrorMessage = "%s label does not match in instancetype: %s"
+	preferenceErrorMessage   = "%s label does not match in preference: %s"
+)
+
+var _ = Describe("Common instance types unit tests", func() {
+
+	skipLabels := map[string]bool{
+		"os-type":    true,
+		"deprecated": true,
+		"version":    true,
+		"class":      true,
+	}
+
+	instanceTypeFunctionMap := map[string]func(string, string, instancetypev1beta1.VirtualMachineClusterInstancetype) error{
+		"cpu":                   checkCPU,
```

Comment:
Match on full label instead of splitting it into parts?

Reply:
Done